app/backend/src/couchers/models.py

```diff
@@ -1029,30 +1037,116 @@ class ClusterEventAssociation(Base):
 
 class Event(Base):
     """
-    A happening
+    An event is compose of two parts:
+
+    * An event template (Event)
+    * An occurence (EventOccurence)
+
+    One-off events will have one of each; repeating events will have one Event,
+    multiple EventOccurences, one for each time the event happens.
     """
 
     __tablename__ = "events"
 
     id = Column(BigInteger, communities_seq, primary_key=True, server_default=communities_seq.next_value())
+    parent_node_id = Column(ForeignKey("nodes.id"), nullable=False, index=True)
 
     title = Column(String, nullable=False)
-    content = Column(String, nullable=False)  # CommonMark without images
+
+    slug = column_property(func.slugify(title))
+
+    creator_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+    owner_user_id = Column(ForeignKey("users.id"), nullable=True, index=True)
+    owner_cluster_id = Column(ForeignKey("clusters.id"), nullable=True, index=True)
     thread_id = Column(ForeignKey("threads.id"), nullable=False, unique=True)
-    geom = Column(Geometry(geometry_type="POINT", srid=4326), nullable=False)
-    address = Column(String, nullable=False)
-    photo = Column(String, nullable=False)
+
+    parent_node = relationship(
+        "Node", backref="child_events", remote_side="Node.id", foreign_keys="Event.parent_node_id"
+    )
+    thread = relationship("Thread", backref="event", uselist=False)
+    subscribers = relationship("User", backref="subscribed_events", secondary="event_subscriptions", lazy="dynamic")
+    organizers = relationship("User", backref="organized_events", secondary="event_organizers", lazy="dynamic")
+    thread = relationship("Thread", backref="event", uselist=False)
+    creator_user = relationship("User", backref="created_events", foreign_keys="Event.creator_user_id")
+    owner_user = relationship("User", backref="owned_events", foreign_keys="Event.owner_user_id")
+    owner_cluster = relationship(
+        "Cluster",
+        backref=backref("owned_events", lazy="dynamic"),
+        uselist=False,
+        foreign_keys="Event.owner_cluster_id",
+    )
+
+    __table_args__ = (
+        # Only one of owner_user and owner_cluster should be set
+        CheckConstraint(
+            "(owner_user_id IS NULL AND owner_cluster_id IS NOT NULL) OR (owner_user_id IS NOT NULL AND owner_cluster_id IS NULL)",
+            name="one_owner",
+        ),
+    )
+
+
+class EventOccurence(Base):
+    __tablename__ = "event_occurences"
+
+    id = Column(BigInteger, communities_seq, primary_key=True, server_default=communities_seq.next_value())
+    event_id = Column(ForeignKey("events.id"), nullable=False, index=True)
+
+    creator_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
```

Comment:
How is creator_user_id different from Event.creator_user_id ? Please write a comment.